include/seastar/coroutine/generator.hh

```diff
@@ -37,11 +37,10 @@
 // asynchronous generation of element sequences.
 namespace seastar::coroutine::experimental {
 
-template<typename Ref, typename Value = void>
-class generator;
-
 namespace internal {
 
+namespace unbuffered {
```

Comment:
nit: some typos in commit message:
> before this change, we only have an unbuffered generator
> implementation, where the producer and consumer operated in a
> "ping-pong" fashion. elements were produced and consumed on at a time.

one at a time

> in this change, we introduce a buffered variant of the generator,
> which allows producer to yield a range of element in batch. it

s/element/elements/

> also enables consumer to process multiple elements without suspension.
> 
> the benefits are:
> 
> * improved efficiency for batch operations
> * reduced context switching between producer and consumer
> * enhanced flexibility in element production and consumption patterns